apps/mobile/src/libs/components/typing-text-view/typing-text-view.tsx

```diff
@@ -0,0 +1,77 @@
+import { Text, View } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	fontWeightToFamily,
+	presetToTextStyle,
+	sizeToTextStyle,
+} from "~/libs/maps/maps";
+import {
+	type StyleProp,
+	type TextStyle,
+	type ValueOf,
+} from "~/libs/types/types";
+
+import {
+	DEFAULT_START_TYPING_DELAY,
+	FIRST_LETTER_INDEX,
+	LETTERS_PRINTED_PER_STEP,
+	NEXT_LETTERS_PRINTED_DELAY,
+} from "./libs/constants/constants";
+import { useTypingText } from "./libs/hooks/hooks";
+
+type Properties = {
+	children: string;
+	lettersPrintedPerStep?: number;
+	maskingColor?: ValueOf<typeof BaseColor>;
+	nextLettersPrintedDelay?: number;
+	preset?: keyof typeof presetToTextStyle;
+	size?: keyof typeof sizeToTextStyle;
+	startTypingDelay?: number;
+	style?: StyleProp<TextStyle>;
+	textColor: ValueOf<typeof BaseColor>;
+	weight?: keyof typeof fontWeightToFamily;
+};
```

Comment:
I believe it can be simplified like this
```suggestion
type Properties = ComponentProps<typeof Text> & {
	lettersPrintedPerStep?: number;
	maskingColor?: ValueOf<typeof BaseColor>;
	nextLettersPrintedDelay?: number;
	startTypingDelay?: number;
	textColor: ValueOf<typeof BaseColor>;
};
```